packages/plugins/plugin-assistant/src/components/Chat/Chat.tsx

```diff
@@ -85,19 +85,22 @@ const ChatRoot = ({ classNames, children, chat, processor, artifact, onEvent, ..
 
   // Messages.
   const queue = useQueue<DataType.Message>(chat?.queue.dxn);
-  const messages = useMemo(
-    () =>
-      dedupeWith(
-        [...(queue?.objects?.filter(Obj.instanceOf(DataType.Message)) ?? []), ...(processor?.messages.value ?? [])],
-        (a, b) => a.id === b.id,
-      ),
-    [queue?.objects, processor?.messages.value],
-  );
+  const pending = useRxValue(processor.messages);
+  const streaming = useRxValue(processor.streaming);
+
+  const messages = useMemo(() => {
+    const queueMessages = queue?.objects?.filter(Obj.instanceOf(DataType.Message)) ?? [];
+    return Result.match(pending, {
+      onInitial: () => queueMessages,
```

Comment:
Can you explain what's the lifecycle of an RX value?

Reply:
the lifecycle is the thing that Tim has said he's not totally happy with yet, but currently they auto-dispose by default unless they are "kept alive" https://github.com/tim-smart/effect-rx/blob/main/packages/rx/test/Rx.test.ts#L32